src/overmapbuffer.h

```diff
@@ -53,6 +55,20 @@ struct city_reference {
     int get_distance_from_bounds() const;
 };
 
+struct camp_reference {
+    static const camp_reference invalid;
+    /** The camp itself, points into @ref overmap::cities */
+    const struct om_camp *camp;
+    /** The global absolute position of the camp (in submap coordinates!) */
```

Comment:
why are you tracking the camp by global omt_pos?  That's the easiest thing to record.